packages/fx-core/src/plugins/resource/frontend/ops/deploy.ts

```diff
@@ -51,60 +52,104 @@ export class FrontendDeployment {
     return lastDeployTime < lastBuildTime;
   }
 
-  public static async doFrontendBuild(
-    componentPath: string,
-    envs: RemoteEnvs,
-    envName: string
-  ): Promise<void> {
+  public static async doFrontendBuild(componentPath: string, envName: string): Promise<void> {
     if (!(await FrontendDeployment.needBuild(componentPath, envName))) {
       return FrontendDeployment.skipBuild();
     }
 
     const progressHandler = ProgressHelper.progressHandler;
 
+    const scripts =
+      (await runWithErrorCatchAndWrap(
+        (error) => new FileIOError(error.message),
+        async () => (await fs.readJSON(path.join(componentPath, PathInfo.NodePackageFile))).scripts
```

Comment:
Nit: Better to wrap the exception of file's nonexistence here to provide user-friendly error message.

Reply:
Do you mean `FileIOError` -> `FileNotFoundError`?  `fs.readJSON` may fail because of invalid json file format, so I think non existence may not be accurate.